docs/source/federation-versions.mdx

```diff
@@ -0,0 +1,471 @@
+---
+title: Federation version changelog
+description: Understand changes between federation versions
+---
+
+## v2.3
+
+**Additions**
+
+* introduce [`@interfaceObject`](https://www.apollographql.com/docs/federation/federated-types/interfaces) directive - a directive that allows you to abstract an interface in isolation by declaring it locally as an object type. This enables you to contribute fields without needing to redefine all the interface’s concrete implementation types.
+
+```graphql
+directive @interfaceObject on OBJECT
+```
+
+`@key` on interface definitions, in order to resolve the implementation type of the abstracted objects returned by the subgraphs using `@interfaceObject`.
+**Note:** This directive was already available in earlier versions of federation, but it wasn't used during composition.
+
+
+## v2.2
+
+**Changes**
+
+Composition now rejects `@shareable` on interface fields. The @shareable directive is about controlling if
+multiple subgraphs can resolve a particular field, and as interface field are never directly resolved (it's their implementation that are), having @shareable on interface fields is not completely meaningful and was never meant to be supported.
+If an existing subgraph does have a @shareable on an interface field, this will now be rejected,
+but the @shareable can simply and safely be removed since it previously was ignored.
+
+The `@shareable` directive has been https://github.com/apollographql/federation/pull/2175. This allows you to use in a type definition and in an extension of that type definition.
+
+```graphql
+directive @shareable repeatable on OBJECT | FIELD_DEFINITION
+```
+
+
+## v2.1
+
+**Additions**
+
+[`@composeDirective`](https://www.apollographql.com/docs/federation/federated-types/federated-directives/#composedirective)
+indicates to composition that all uses
+of a particular custom type system directive in the subgraph schema should be preserved
+in the supergraph schema (by default, composition omits most directives from the supergraph schema).
+
+```graphql
+directive @composeDirective(name: String!) repeatable on SCHEMA
+```
+
+Here's an example of how to use `@composeDirective`:
```

Comment:
you provide an example here on how to use compose directive? should we have similar example for `@interfaceObject`?